app/Http/Controllers/StationController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers;
+
+use App\Station;
+use App\Services\StationService;
+use App\Http\Requests\StoreStation;
+use App\Http\Requests\UpdateStation;
+use Illuminate\Http\Response;
+use Illuminate\Http\Request;
+use Illuminate\Http\JsonResponse;
+
+class StationController extends Controller
+{
+    public function __construct(StationService $station)
+    {
+        $this->station = $station;
+    }
+    /**
+     * Display all resources
+     */
+    public function index(): JsonResponse
+    {
+        $stations = Station::all();
+        return response()->json([$stations], Response::HTTP_CREATED);
+    }
+    /**
+     * Store a newly created resource in storage.
+     * @param  App\Http\Requests\StoreStation $request
+     */
+    public function store(StoreStation $request): JsonResponse
+    {
+        Station::store($request->validated());
+        return response()->json(['message' => 'Station created succesfully'], Response::HTTP_CREATED);
+    }
+
+    /**
+     * Display the specified resource.
+     * @param  \App\Station  $station
+     */
+    public function show(Station $station): JsonResponse
+    {
+        return response()->json($station, Response::HTTP_OK);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     * @param  App\Http\Requests\UpdateStation  $request
+     * @param  \App\Station  $station
+     */
+    public function update(UpdateStation $request, Station $station): JsonResponse
+    {
+        $station->update($request->validated());
+        return response()->json(['message' => 'Station updated succesfully'], Response::HTTP_OK);
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     * @param  \App\Station  $station
+     */
+    public function destroy(Station $station): JsonResponse
+    {
+        $station->delete();
+        return response()->json(['message' => 'Station deleted successfully'], Response::HTTP_OK);
+    }
+
+    /**
+     * Example of fake (won't be implemented) method that should utilize external service
+     * because the query might be more complicated and it interacts with User model
+     * We need to discuss how to handle filtering yet...
+     * @param Request $request
+     * @return void
+     */
+    public function subscribed(): JsonResponse
+    {
+        //$this->station is an instance of StationService
+        $stations = $this->station->subscribed();
```

Comment:
And I think I don't understand why sometimes you're using service for retrieving data and sometimes you're using Eloquent. Plus in this case you're attaching service to every call to every method in this controller; you can move this injection to method itself.

Reply:
My thought process is that this kind of approach is a middle ground between full-service approach and having fat controllers. 
By using Eloquent I'm already using an additional layer (similar to what a Repository would be) between the controller and the database, but then again I think more complicated queries (or worse, the code containing business logic) should not be placed inside of it, thus the dilemma.
 
Using repository pattern or mirroring all controller methods in Services just for executing simple Eloquent queries sounds like a consistent resolution, but might go against KISS and YAGNI approaches.

Agreed on the method injection. Initially I wanted to inject full crud repository, but I dropped that approach.